service/controller/v2/resource/node/create.go

```diff
@@ -114,30 +123,58 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 
 	if len(customPods) > 0 {
 		r.logger.LogCtx(ctx, "level", "debug", "message", "deleting all pods running custom workloads")
+		t1 := time.Now()
 
+		var wg sync.WaitGroup
 		for _, p := range customPods {
-			err := k8sClient.CoreV1().Pods(p.GetNamespace()).Delete(p.GetName(), &apismetav1.DeleteOptions{})
-			if err != nil {
-				return microerror.Mask(err)
-			}
+			// delete each pod in separate goroutine to avoid blocking
+			wg.Add(1)
+			go func(k8sClient kubernetes.Interface, pod v1.Pod) {
+				defer wg.Done()
+				o := func() error {
+					return EvictPod(k8sClient, pod)
```

Comment:
Please always do explicit error handling. This code hides the trace of the error now, which makes debugging harder.